### PR TITLE
Make Version class inheritable

### PR DESCRIPTION
**Pull request by joint-tech**

### CODE REVIEW

File: src/Version.php [hunk 1]
```diff
@@ -40,7 +40,21 @@ class Version
...
+    /**
+     * Init version object.
+     * 
+     * @param string $version Version string
+     * 
+     * @return void 
+     * 
+     * @throws \PHLAK\SemVer\Exceptions\InvalidVersionException 
+     */
+    protected function init(string $version): void
```
Comment (PHLAK) [inline src/Version.php:57]: What's the purpose of this change?

Comment (joint-tech) [reply src/Version.php:57]: I modified the parse method to create an instance of the child class.
As pointed out by Psalm, using `new static` may lead to an [unsafe instantiation](https://psalm.dev/docs/running_psalm/issues/UnsafeInstantiation/) because the constructor can change in the child class.
By making the constructor final it forces the use of the parent constructor. By adding the init method it still gives the ability to perform operations during instantiation.

Comment (PHLAK) [reply src/Version.php:57]: I'd prefer to keep constructor _not_ final (and remove the `init()` method).

I'll with static analysis errors if those are the only things keeping this from passing CI.
